src/gui.js

```diff
@@ -223,10 +223,11 @@ export default class GUI {
             });
 
             component.on('input', (data) => {
+                let oldValue = opts.object[opts.property];
                 if(opts.object && opts.property)
                     opts.object[opts.property] = data;
 
-                if(opts.onChange) {
+                if(data !== oldValue && opts.onChange) {
```

Comment:
IMO, deciding whether to emit a change should be done at the component level, not here (I think you've accomplished this with your changes to the text component). With this change, if a user wants to write a text component that has different behavior, they'd also have to adjust the behavior here.